vllm_spyre/v1/worker/spyre_model_runner.py

```diff
@@ -612,13 +613,26 @@ def _update_states(self, scheduler_output):
             del self.requests[req_id]
             self.input_batch.remove_request(req_id)
 
+        # free the blocks used for padding to minimum decode batch size of 2
+        if not scheduler_output.total_num_scheduled_tokens \
+                and self.dummy_req_ids2blocks:
+            for freed_block in self.dummy_req_ids2blocks:
+                self.free_blocks.append(freed_block)
+            self.dummy_req_ids2blocks = []
+
     def _prepare_prompt(
         self,
         new_requests: list[NewRequestData],
     ) -> ModelForwardInputs:
         assert len(new_requests) > 0
         input_token_list: list[torch.Tensor] = []
 
+        # free the blocks used for padding to minimum decode batch size of 2
+        if self.dummy_req_ids2blocks:
+            for freed_block in self.dummy_req_ids2blocks:
+                self.free_blocks.appendleft(freed_block)
```

Comment:
I think `appendleft` won't do exactly what you want. It will append the values from `dummy_req_ids2blocks` to the left, but in reverse order. e.g. `[6, 7] --> appendleft --> [7, 6, 8, 9, 10, ...]`. I would suggest to use `append` simply, just as you did in line 620. Or iterate the `dummy_req_ids2blocks` in reverse order (and do the same in 620)